code/z_adventurezones/lavamoon.dm

```diff
@@ -883,12 +896,18 @@ var/global/iomoon_blowout_state = 0 //0: Hasn't occurred, 1: Moon is irradiated
 				if (spawners)
 					for (var/obj/iomoon_boss/bot_spawner/aSpawner in spawners)
 						aSpawner.spawn_bot()
-				if (rotors)
-					rotors.icon_state = "powercore_rotors_fast"
+				if (base || iomoon_blowout_state == 1)
+					base.icon_state = "powercore_base_fast"
+			top.icon_state = "powercore_core_hit"
+			SPAWN(0.6 SECONDS)
+				if (IOCORE_ONLINE && src.health <= PANIC_HEALTH_LEVEL)
+					top.icon_state = "powercore_core_fast"
+				else if (IOCORE_ONLINE)
+					top.icon_state = "powercore_core_active"
```

Comment:
```suggestion
			if (!ON_COOLDOWN(top, "hit_effect", 0.7 SECONDS))
				top.icon_state = "powercore_core_hit"
				SPAWN(0.6 SECONDS)
					if (IOCORE_ONLINE && src.health <= PANIC_HEALTH_LEVEL)
						top.icon_state = "powercore_core_fast"
					else if (IOCORE_ONLINE)
						top.icon_state = "powercore_core_active"
```

Reply:
Also the `if (IOCORE_ONLINE` should be updated to use more defines as mentioned in the next review comment.